Add sriov chart (#1211)
rancher/charts: packages/rancher-sriov/generated-changes/overlay/app-README.md

# Rancher SR-IOV Network Operator

> This chart is based on the upstream [k8snetworkplumbingwg/sriov-network-operator](https://github.com/k8snetworkplumbingwg/sriov-network-operator) project. The chart deploys the SR-IOV Operator and its CRDs, which is designed to help the user provision and configure the SR-IOV CNI in a cluster that uses [Multus CNI](https://github.com/k8snetworkplumbingwg/multus-cni), to provide high performing extra network interfaces to pods. This chart is expected to be deployed on an RRKE2 cluster and only meant for advanced use cases where multiple CNI plugins and high performing network interfaces on pods are required. Users who do not need these features are not advised to install this chart.

Reviewer: nit:
```suggestion
This chart is based on the upstream [k8snetworkplumbingwg/sriov-network-operator](https://github.com/k8snetworkplumbingwg/sriov-network-operator) project. The chart deploys the SR-IOV Operator and its CRDs, which are designed to help the user provision and configure the SR-IOV CNI in a cluster that uses [Multus CNI](https://github.com/k8snetworkplumbingwg/multus-cni), to provide high performing extra network interfaces to pods. This chart is expected to be deployed on an RKE2 cluster and only meant for advanced use cases where multiple CNI plugins and high performing network interfaces on pods are required. Users who do not need these features are not advised to install this chart.
```

Author: Thanks

The chart installs the following components:

>  - SR-IOV Operator - An operator that helps provision and configure the SR-IOV CNI plugin and SR-IOV Device plugin (requires Multus)

Reviewer: nit: `requires Multus` is also mentioned above. Do we need this?
```suggestion
 - SR-IOV Operator - An operator that helps provision and configure the SR-IOV CNI plugin and SR-IOV Device plugin
```

# Rancher SR-IOV Network Operator

> This chart is based on the upstream [k8snetworkplumbingwg/sriov-network-operator](https://github.com/k8snetworkplumbingwg/sriov-network-operator) project. The chart deploys the SR-IOV Operator and its CRDs. It allows users to deploy SR-IOV CNI, which can be used, together with [Multus](https://github.com/k8snetworkplumbingwg/multus-cni), to provide high performing extra network interfaces to pods.

Reviewer: ```suggestion
This chart is based on the upstream [k8snetworkplumbingwg/sriov-network-operator](https://github.com/k8snetworkplumbingwg/sriov-network-operator) project. The chart deploys the SR-IOV Operator and its CRDs, which is designed to help the user to provision and configure SR-IOV CNI and Device plugins in a cluster that uses [Multus CNI](https://github.com/k8snetworkplumbingwg/multus-cni) to provide high performing extra network interfaces to pods.
```

> This chart is based on the upstream [k8snetworkplumbingwg/sriov-network-operator](https://github.com/k8snetworkplumbingwg/sriov-network-operator) project. The chart deploys the SR-IOV Operator and its CRDs, which is designed to help the user provision and configure the SR-IOV CNI in a cluster that uses [Multus CNI](https://github.com/k8snetworkplumbingwg/multus-cni), to provide high performing extra network interfaces to pods.

Reviewer: Based on offline conversation, we need to make sure this file calls out that:
1) This chart is expected to be deployed on an RKE2 cluster. It has not been tested or designed for any other cluster type at the moment, although it can be deployed in other clusters with some modifications. (Addresses https://github.com/rancher/charts/pull/1211#issuecomment-853288767)
2) A user who is deploying this chart is expected to be fairly knowledgable about Multus, SR-IOV, and the use case behind this chart as it is only meant for **advanced use cases** where multiple CNIs and high performing network interfaces on pods are required. Users who do not need these features are not advised to install this chart.

cc: @cjellick @cbron

Reviewer 2: In the case of the sriov chart, although we can fix the pod tolerations to make it installed successfully in a rke1 cluster,  I think we will keep it as it is today so the failure can also bring the user’s attention and realize the cluster is not an RKE2 cluster.

The chart installs the following components:

>  - SR-IOV Operator - The operator helps provision and configure SR-IOV CNI plugin and Device plugin. It requires Multus to operate correctly.
>  - SR-IOV Network Config Daemon - Deployed by the operator, this daemon is responsible for discovering the SR-IOV NICs on each node

Reviewer: nit: even if you don't take these changes, make sure you either add punctuation to the second bullet point or remove it from the first.

```suggestion
 - SR-IOV Operator - An operator that helps provision and configure CNI or Device plugins (requires Multus)
 - SR-IOV Network Config Daemon - A Daemon deployed by the Operator that discovers SR-IOV NICs on each node
```

Author: I partially applied your suggestion. I think we should have `provision and configure SR-IOV CNI plugin and SR-IOV Device plugin`. If we remove SR-IOV, the user could think that any CNI or any Device Plugin can be deployed.

What do you mean by adding punctuation to the second bullet?

Reviewer: You resolved the issue. I mean that the first had a `.` at the end and the second did not 😄
